### ISSUE TITLE
Implement a way to stream run highlevel backtest by processing each data config in each run

### ISSUE THREAD

**Issue by TheLegendAli** [nautechsystems/nautilus_trader#1943]
# Feature Request

For backtesting tick data on longer time frames, there is a lot of data to feed inside a ram. Currently, we can stream the backtest through chunks, I would like to do it through time. In backtest we can divide the data by time or quarter(for example we create a data config for each front future contract) then we pass the list of data configs to the backtest engine, which streams the data in each data config.

**Comment by cjdsellers** [comment]
Hey @TheLegendAli

Thanks for your thoughts here.

I'm interested to understand more deeply why you'd like to stream based on time, I'm not sure there would be any material change to the streaming where the intent is to limit the amount of memory being used by data?

I'm concerned we'd be introducing another configuration option and increasing complexity unnecessarily?

Also note [this commit](https://github.com/nautechsystems/nautilus_trader/commit/795058479272a7a07f955a2571f3659960ba2fbb) which aligned and clarified the `chunk_size` terminology.

**Comment by TheLegendAli** [comment]
Hi Chris,

Few thoughts:

1. I wrote this before the commit, but I noticed that streaming by chunks would take more ram than one shot(for example if one shot would take 5GB, streaming would take 10). I'm not sure why that's happening.

2. I want to stream data but keep the efficiency. For my use case, since I am backtesting on future contracts that expire every 3 months, it made sense to divide them into 3 months and stream based on time.

3. Since I wrote this, decided to put it out there in case someone else wants to use this approach.

that being said I sympathize with your concern about complexity and understand why you might not want to merge this.
